src/views/AnalysisView.vue

```diff
@@ -0,0 +1,298 @@
+  },
+  methods: {
+    getData: async function () {
+      this.fetching = true;
+      // Extract query information
+      const where = {
+        type: this.type,
+        recordingDateTime: this.dateQuery()
+      };
+      const limit = 1000;
+      const params = {
+        where: JSON.stringify(where),
+        limit: limit,
+        offset: 0,
+        tagMode: "any"
+      };
+
+      // Get all data (first 1000 rows)
+      let allData = await api.query(params);
+      // Check whether all data was fetched
+      // if not, run again with increased limit to get all rows
+      if (allData.count > limit) {
+        params.limit = allData.count;
+        allData = await api.query(params);
+      }
+      // Count the number of recordings for each device
+      this.devices.map((device) => this.deviceCount[device.id] = 0);
+      for (const row of allData.rows) {
+        this.deviceCount[row.Device.id] += 1;
+      }
+      // Create data and label variables
+      const labels = [];
+      const data = [];
+      this.unused = [];
+      for (const device of this.devices) {
+        if (this.deviceCount[device.id] > 0) {
+          data.push({
+            id: device.id,
+            count: this.deviceCount[device.id],
+            devicename: device.name
+          });
+          labels.push(device.name);
+        } else {
+          this.unused.push(device.name);
+        }
+      }
+
+      // Create colors for bar graphs
+      const colorPicker = () => {
+        let hue;
+        if (this.lastHue < 360) {
+          hue = this.lastHue + 60;
+          this.lastHue = hue;
+        } else {
+          hue = this.lastHue - 339;
+          this.lastHue = hue;
+        }
+        const hsl = `hsl(${hue}, 80%, 80%)`;
+        return hsl;
+      };
+      this.lastHue = -60; // reset starting hue
+      const colors = data.map(() => colorPicker());
+      // Create dataset suitable for ChartJS
+      this.data = {
+        labels: labels,
+        datasets: [
+          {
+            data: data.map((item) => item.count),
+            backgroundColor: colors,
+            borderColor: colors,
+            borderWidth: 1
+          }
+        ]
+      };
+      const title = 'Device Activity';
+      if (this.dateRange === 0) {
+        this.title = `${title} (All time)`;
+      } else if (this.dateRange ===1) {
+        this.title = `${title} (Last 24 Hours)`;
+      }else {
+        this.title = `${title} (Last ${this.dateRange} days)`;
+      }
+      this.fetching = false;
+    },
+    dateQuery() {
+      const parseDate = function (date) {
+        const hour = (0 + date.getHours().toString()).slice(-2);
+        const min = (0 + date.getMinutes().toString()).slice(-2);
+        const day = (0 + date.getDate().toString()).slice(-2);
+        const month = (0 + (date.getMonth() + 1).toString()).slice(-2);
+        const year = date.getFullYear();
+        return year + "-" + month + "-" + day + " " + hour + ":" + min + ":" + "00";
```

Comment:
I suspect Date's `toISOString` method will provide exactly the right kind of date string for the API, avoiding the need for parseDate. Just make sure the API server handles the time zone offset correctly though... (it should)